### ISSUE TITLE
CompleteWithinAsync with AssertionScope will wait task to return value

### ISSUE THREAD

**Issue by xzxzxc** [fluentassertions/fluentassertions#2004]
### Description

In the case of using `AssertionScope` with the `CompleteWithinAsync` method over the delegate that returns a task with a result (`Task<T>`), the method will wait until the task is completed.

### Complete minimal example reproducing the issue

```csharp
using var _ = new AssertionScope();
var someAsyncWork = async () =>
{
    while (true)
        await Task.Delay(20.Milliseconds());

    return 1;
};

await someAsyncWork.Should().CompleteWithinAsync(500.Milliseconds());
```

### Expected behavior:

The `CompleteWithinAsync` method will throw an exception in a decent period of time.

### Actual behavior:

Execution stucks in the infinite loop.

### Versions

I'm using FluentAssertions 6.7.0 with net6.0.

### Additional Information

Issue disappears with one of the following changes:
 - remove usage of the `AssertionScope`;
 - remove `return` statement from the  `someAsyncWork` (change the type of the delegate from `Func<Task<T>>` to `Func<Task>`).

**Comment by lg2de** [comment]
@dennisdoomen, I'll take care. Please assign this issue to me.

**Comment by xzxzxc** [comment]
@lg2de thank you!
I can confirm, the issue is fixed in the 6.8.0.
